### DISCUSSION TITLE
Delete a subscription from a connection

### DISCUSSION THREAD

**Discussion by ton3r** [emqx/MQTTX#1636] [Q&A]
It seems that I need to be connected to mqtt to remove a subscription.

But I have now the Problem that I am not allowed to subscribe to the subscription anymore and the client disconnect. But then I am not able to remove the subscription because it says "not connected"

How to force this?

--Tom

**Comment by ysfscream** [comment]
hi, tom. thanks for your feedback, and sorry for my delayed reply.

We haven't encountered this problem before, and it seems there is no way to force a deletion, as the subscription and unsubscription in MQTTX are akin to adding and deleting in a database. If this problem is common, we will consider supporting the deletion of topics directly from the database offline, not just upon unsubscription.

**Reply by ton3r** [reply]
Actual I have to delete the complete Connection Settings, to create the same without the "dead" subscription. Because when I (re) connect to the connection config, it directly tries to subscribe to the topic where I am not allowed anymore to subscribe.

**Reply by ysfscream** [reply]
YES, and if you can, clicking the `duplicate` button will quickly create the same connection without the subs.

<img width="357" alt="image" src="https://github.com/emqx/MQTTX/assets/21299158/3b09bec9-c811-4cfd-bf0b-f8118253d51b">
